### PR TITLE
DO-673 Adding buildspec.yml to run CI pipeline

### PR DESCRIPTION
**Pull request by marcelomilera**
- Adding buildspec.yml to run CI pipeline
- Updating Dockerfile to reduce docker image size (from ~700MB to ~300MB)

### CODE REVIEW

File: buildspec.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+env:
+  git-credential-helper: yes
+  parameter-store:
+    DOCKER_HUB_USER: '/devops/shared/DOCKER_HUB_USER'
+    DOCKER_HUB_PASSWORD: '/devops/shared/DOCKER_HUB_PASSWORD'
+
+phases:
+  install:
+    runtime-versions:
+      nodejs: 16
```
Comment (Arthaclarius) [inline buildspec.yml:12]: Is there a reason we use node 16 here and node 18 in the docker?

File: buildspec.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    commands:
+      - n 18
+      - npm update -g npm
+      - npm install -g yarn
```
Comment (Arthaclarius) [inline buildspec.yml:16]: Maybe I'm wrong, but isn't yarn preinstall with node>16?

https://yarnpkg.com/getting-started/install#nodejs-1610

**Review by Arthaclarius — CHANGES_REQUESTED** [review]

File: buildspec.yml (hunk 1 above)
Comment (marcelomilera) [reply buildspec.yml:12]: Node 16 is the latest available runtime for AWS CodeBuild hosts. But we can manually install Node 18 by running `n 18` (line 14).
I tried to keep Node 18 as it was the version used in the Dockerfile.

File: buildspec.yml (hunk 2 above)
Comment (marcelomilera) [reply buildspec.yml]: you're right. I removed this line.

**Review by Arthaclarius — APPROVED** [review]

**Pull request merged**